src/js/media-controller.js

```diff
@@ -468,6 +466,10 @@ const shouldGetMediaUIElementDescendants = (el) => {
  *  - Have a `media-chrome-attributes` attribute with at least one well-defined media chrome attribute
  */
 const getMediaUIElementDescendants = (rootNode) => {
+  // If the rootNode is a custom element that's not yet defined/ready, so return an array of a promise to yield the
+  // element once it's registered for us to check at that time.
+  if (isUndefinedCustomElement(rootNode)) return [getRegisteredCustomElementPromise(rootNode)];
```

Comment:
Returning a promise that we then need to check for later is creating some pretty complicated logic. Can we instead check the element defined-state when we need to operate on it and defer that operation then?

Reply:
If I'm understanding your proposal correctly, we'd want to make sure any "not yet registered but eventually will be identified as media ui elements" are "initialized" correctly with any values, and we'd (probably) want to **_not_** "associate" "not yet registered but eventually but eventually will definitely not be identified as media ui elements", but we can talk through alternative ways of handling these cases when we go through the code together.